components/webgpu/lib.rs

```diff
@@ -475,9 +474,7 @@ impl<'a> WGPU<'a> {
                             ))
                             .map_err(|e| format!("{:?}", e))
                         };
-                        if result.is_err() {
-                            self.encoder_record_error(command_encoder_id, result.clone());
-                        }
+                        self.encoder_record_error(command_encoder_id, result.clone());
```

Comment:
instead of cloning the error unconditionally, we should probably pass it by reference and clone inside the function only if needed